src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx

```diff
@@ -145,8 +145,11 @@ export default class IntegrationRepos extends AsyncComponent {
         </DropdownButton>
       );
     }
-    let repositories = this.state.integrationRepos.repos;
-    let items = (repositories || []).map(repo => {
+    const repositories = new Set(this.state.itemList.map(i => i.name));
+    let repositoryOptions = this.state.integrationRepos.repos.filter(repo => {
+      return !repositories.has(repo.identifier);
+    });
+    let items = (repositoryOptions || []).map(repo => {
```

Comment:
With the guard above, this can just be `repositoryOptions` since it'll always be an array.